compiler/src/iree/compiler/Codegen/Common/EncodingUtils.cpp

```diff
@@ -70,23 +69,19 @@ static RankedTensorType transposeIfNarrowNResult(RankedTensorType tensorType) {
   for (auto &map : maps) {
     map = map.compose(permutation);
   }
-  SmallVector<Attribute> newMaps;
-  for (auto map : maps) {
-    newMaps.push_back(AffineMapAttr::get(map));
-  }
-  ArrayAttr newIndexingMaps = ArrayAttr::get(context, newMaps);
   auto elemType = tensorType.getElementType();
-  OpBuilder builder(context);
+  auto operandIndex = encoding.getOperandIndex().getInt();
 
-  auto opTypeAttr = IREE::Encoding::EncodingOpTypeAttr::get(
-      context, IREE::Encoding::EncodingOpType::matmul);
   // TODO(#17718): Handle the broadcast map for transpose cases. It is on the
   // experimental path, so it is not clear what needs to be done here. For now
   // just use the original map for the new encoding.
+  std::optional<AffineMap> newBcastMap;
+  if (encoding.getBcastMap()) {
+    newBcastMap = encoding.getBcastMap().getValue();
+  }
```

Comment:
This is the only part of this PR that, instead of simplifying code, adds more complex code. Why can't we continue to just pass `encoding.getBcastMap()` to `IREE::Encoding::EncodingAttr::get` like in the existing code?

Reply:
Good question. This is because `encoding.getBcastMap()` returns an attribute (i.e., AffineMapAttr), while the builder requires C++ structured/builtin types (i.e., `std::optional<AffineMap>`). Instead of having the logic here, I think we can add an `std::optional<AffineMap> getBcastMapValue()` method to the encoding attribute. What do you think?

Reply:
I see. I like the idea, but really don't favor adding `extraClassDeclarations`. Can this be a plain global C++ function instead (not declared in .td, not member)? Otherwise, feel free to just drop that commit, I don't want to make this PR hard to merge and I liked the original form very much already.

Reply:
I can try to put it to a better state when I start poking the broadcast map things. I don't really know what methods could be useful and we can revisit later (when we have more needs). Revert the commits for now.